dns/jsonify/main.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/kubernetes/test/e2e/perftype"
+)
+
+const (
+	// S_TO_MS is a second to milisecond ratio.
+	S_TO_MS = 1000
+)
+
+// BenchamrkResult is a dns benchmark results structure.
+type BenchamrkResult struct {
+	Code   int                    `json: code`
+	Data   map[string]interface{} `json: data`
+	Params map[string]interface{} `json: params`
+}
+
+func main() {
+	err := run()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func run() error {
+	var benchmarkDirPath string
+	var jsonDirPath string
+	var benchmarkName string
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	flag.StringVar(&benchmarkDirPath, "benchmarkDirPath", ".", "benchmark results directory path")
+	flag.StringVar(&jsonDirPath, "jsonDirPath", ".", "json results directory path")
+	flag.StringVar(&benchmarkName, "benchmarkName", ".", "benchmark name")
+
+	if err := flag.CommandLine.Parse(os.Args[1:]); err != nil {
+		return fmt.Errorf("flag parse failed: %v", err)
+	}
+
+	fmt.Printf("benchmarkDirPath: %v\n", benchmarkDirPath)
+	fmt.Printf("jsonDirPath: %v\n", jsonDirPath)
+	fmt.Printf("benchmarkName: %v\n", benchmarkName)
+
+	latency := perftype.PerfData{Version: "v1"}
+	latencyPerc := perftype.PerfData{Version: "v1"}
+	queries := perftype.PerfData{Version: "v1"}
+
+	fileList, err := getFileList(benchmarkDirPath)
+	if err != nil {
+		return fmt.Errorf("listing files error: %v", err)
```

Comment:
don't wrap errors unless there is a good reason - just `return err`

Reply:
> don't wrap errors unless there is a good reason - just return err

I though wrapping errors is common practice in golang - it allows to identified where the error came from. Assuming that you are not a fan of this approach, can you tell me what are the drawback of wrapping errors?